base_layer/core/src/transactions/transaction_protocol/single_receiver.rs

```diff
@@ -108,22 +108,25 @@ impl SingleReceiverTransactionProtocol {
                 .range_proof
                 .construct_proof(&spending_key, sender_info.amount.into())?
         };
-        Ok(TransactionOutput::new(
+        let output = TransactionOutput::new(
             features,
             commitment,
             RangeProof::from_bytes(&proof)
                 .map_err(|_| TPE::RangeProofError(RangeProofError::ProofConstructionError))?,
             sender_info.script.clone(),
             sender_info.script_offset_public_key.clone(),
-        ))
+            sender_info.sender_metadata_signature.clone(),
+        );
+        let _ = output.verify_sender_signature()?;
```

Comment:
Doesn't this method return a unit and therefore can be ignored? `()`
There was another instance of this somewhere else as well. Perhaps an artefact of debugging?
```suggestion
        output.verify_sender_signature()?;
```